Source/SelfService/Web/microservice/dataTable.tsx

```diff
@@ -0,0 +1,211 @@
+            outline: 'none'
+        },
+        '.MuiIconButton-root': {
+            'visibility': 'visible',
+            ':hover': {
+                backgroundColor: 'transparent'
+            }
+        },
+        '.MuiDataGrid-sortIcon': {
+            color: 'rgba(255, 255, 255, 0.38);',
+            width: '20px',
+            height: '20px'
+        },
+        '.MuiDataGrid-columnHeader:not(.MuiDataGrid-columnHeader--sorted) .MuiDataGrid-sortIcon': {
+            opacity: 1
+        }
+    },
+    status: {
+        display: 'flex',
+        justifyContent: 'center'
+    },
+    statusTitle: {
+        fontWeight: '500',
+        fontSize: '12px',
+        lineHeight: '22px',
+        letterSpacing: '0.06em',
+        textTransform: 'uppercase',
+        marginInlineStart: '10px'
+    }
+};
+
+export type MicroserviceObject = {
+    id: string,
+    name: string,
+    kind: string,
+    environment: string,
+    live: MicroserviceInfo,
+    edit: {},
+    status?: []
+};
+
+type DataTableProps = {
+    environment: string,
+    application: HttpResponseApplication,
+    microservices: MicroserviceObject[]
+};
+
+export const DataTable: React.FC<DataTableProps> = ({ application, environment, microservices }: DataTableProps) => {
+    const history = useHistory();
+    const [rows, setRows] = useState<(MicroserviceObject | undefined)[]>([]);
+
+    useEffect(() => {
+        const mapStatusToMicroservicesAndSetState = async () => {
+            const rows = await Promise.all(microservices.map(async microservice => {
+                const status = await getMicroserviceStatus(microservice.id);
+
+                return {
+                    ...microservice,
+                    status
+                } as MicroserviceObject;
+            }));
+
+            setRows(rows);
+        };
+
+        mapStatusToMicroservicesAndSetState();
+    }, [setRows]);
+
+    const getMicroserviceStatus = (microserviceId: string) => {
+        return new Promise((resolve, reject) => {
+            resolve(
+                getPodStatus(application.id, environment, microserviceId)
+                    .then(res => res.pods)
+                    .catch(err => reject(err))
+            );
+        });
+    };
+
+    const columns: GridColDef[] = [
+        {
+            field: 'name',
+            headerName: 'Name',
+            minWidth: 200,
+            flex: 1
+        },
+        {
+            field: 'image',
+            headerName: 'Container Image',
+            minWidth: 200,
+            flex: 1,
+            valueGetter: (params: GridValueGetterParams) =>
+                `${params.row.edit.extra.headImage || ''}`,
+        },
+        {
+            field: 'runtime',
+            headerName: 'Runtime',
+            minWidth: 200,
+            flex: 1,
+            valueGetter: (params: GridValueGetterParams) => {
+                const runtimeNumber = params.row.edit.extra.runtimeImage.replace(/[dolittle/runtime:]/gi, '');
```

Comment:
I am pretty sure the regex here needs to be `/dolittle\/runtime:/gi` instead of `/[dolittle/runtime:]/gi`. If you need a place to try it out, http://regex101.com is a great page.